File: SAML 2 IdP-Initiated Web Example for ASP.NET MVC/[VB]-SAML 2 IdP-Initiated Web Example for ASP.NET MVC/VB/Saml2IdpInitiatedMvc.IdP/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace Saml2IdpInitiatedMvc.Controllers
	<HandleError> _
	Public Class HomeController
		Inherits Controller
		<Authorize> _
		Public Function Index() As ActionResult
			ViewData("Message") = "Welcome to SAML SSO Idp-Initiated IdP Example for ASP.NET MVC!"

			Dim serviceProviderUrl As String = System.Web.Configuration.WebConfigurationManager.AppSettings("ServiceProviderUrl")
			ViewData("NavigateUrl") = System.Web.VirtualPathUtility.ToAbsolute(String.Format("~/Service?spUrl={0}", HttpUtility.UrlEncode(serviceProviderUrl)))

			Return View()
		End Function

		Public Function About() As ActionResult
			Return View()
		End Function
	End Class
End Namespace

File: SAML 2 IdP-Initiated Web Example for ASP.NET MVC/[C#]-SAML 2 IdP-Initiated Web Example for ASP.NET MVC/C#/Saml2IdpInitiatedMvc.IdP/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace Saml2IdpInitiatedMvc.Controllers
{
    [HandleError]
    public class HomeController : Controller
    {
        [Authorize]
        public ActionResult Index()
        {
            ViewData["Message"] = "Welcome to SAML SSO Idp-Initiated IdP Example for ASP.NET MVC!";

            string serviceProviderUrl = System.Web.Configuration.WebConfigurationManager.AppSettings["ServiceProviderUrl"];
            ViewData["NavigateUrl"] = System.Web.VirtualPathUtility.ToAbsolute(string.Format("~/Service?spUrl={0}", HttpUtility.UrlEncode(serviceProviderUrl)));

            return View();
        }

        public ActionResult About()
        {
            return View();
        }
    }
}
